hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionChecker.java

```diff
@@ -43,16 +51,39 @@
  */
 public class EvictionChecker {
 
-    protected static final int ONE_HUNDRED_PERCENT = 100;
-    protected static final int ONE_KILOBYTE = 1024;
-    protected static final int ONE_MEGABYTE = ONE_KILOBYTE * ONE_KILOBYTE;
+    protected static final String HAZELCAST_MEMORY_INFO_ACCESSOR_IMPL = "hazelcast.memory.info.accessor.impl";
+    protected static final double ONE_HUNDRED_PERCENT = 100D;
 
+    protected final ILogger logger;
     protected final MapServiceContext mapServiceContext;
     protected final MemoryInfoAccessor memoryInfoAccessor;
 
-    public EvictionChecker(MemoryInfoAccessor memoryInfoAccessor, MapServiceContext mapServiceContext) {
-        this.memoryInfoAccessor = memoryInfoAccessor;
+    public EvictionChecker(MemoryInfoAccessor givenMemoryInfoAccessor, MapServiceContext mapServiceContext) {
+        checkNotNull(givenMemoryInfoAccessor, "givenMemoryInfoAccessor cannot be null");
+        checkNotNull(mapServiceContext, "mapServiceContext cannot be null");
+
+        this.logger = mapServiceContext.getNodeEngine().getLogger(getClass());
         this.mapServiceContext = mapServiceContext;
+
+        MemoryInfoAccessor pluggedMemoryInfoAccessor = getPluggedMemoryInfoAccessor();
```

Comment:
shouldn't we move the custom `MemoryInfoAccessor` creation into a whoever is creating the `EvictionChecker` ?